crates/dojo-core/tests/src/world.cairo

```diff
@@ -450,6 +544,49 @@ fn test_revoke_resource() {
     assert(*status[0] == 0, 'access not revoked');
 }
 
+#[test]
+#[available_gas(9000000)]
+fn test_assume_admin_role_by_admin() {
+    // Spawn empty world
+    let world = spawn_empty_world();
+
+    // No Auth route
+    let mut route = ArrayTrait::new();
+
+    // Initialize world
+    world.initialize(route);
+
+    // Assume Admin role by Admin
+    let mut components = ArrayTrait::<ShortString>::new();
+    world.assume_role(World::ADMIN.into(), components);
+
+    // Check that role is assumed
+    assert(world.execution_role() == World::ADMIN.into(), 'role not assumed');
+}
+
+#[test]
+#[available_gas(9000000)]
+#[should_panic]
+#[ignore] // TODO:: Remove once set_caller_address is working properly
```

Comment:
`starknet::testing::set_caller_address` doesn't seem to work properly so I marked this as ignored for now